src/mprk.jl

```diff
@@ -501,6 +501,10 @@ function alg_cache(alg::MPRK22, u, rate_prototype, ::Type{uEltypeNoUnits},
                    ::Type{uBottomEltypeNoUnits}, ::Type{tTypeNoUnits},
                    uprev, uprev2, f, t, dt, reltol, p, calck,
                    ::Val{true}) where {uEltypeNoUnits, uBottomEltypeNoUnits, tTypeNoUnits}
+    if !(f isa PDSFunction || f isa ConservativePDSFunction)
+        throw(ArgumentError("MPRK22 can only be applied to production-destruction systems"))
+    end
+    @assert alg.alpha≥1 / 2 "MPRK22 requires α ≥ 1/2."
```

Comment:
```suggestion
    if !(alg.alpha ≥ 1 / 2)
        throw(ArgumentError("MPRK22 requires α ≥ 1/2."))
    end
```

From the docstring:
```julia
help?> @assert
  @assert cond [text]

  Throw an AssertionError if cond is false. Preferred syntax for writing assertions. Message text
  is optionally displayed upon assertion failure.

  │ Warning
  │
  │  An assert might be disabled at various optimization levels. Assert should therefore
  │  only be used as a debugging tool and not used for authentication verification (e.g.,
  │  verifying passwords), nor should side effects needed for the function to work
  │  correctly be used inside of asserts.

```

Thus, I think it's better to throw a proper error in this case.